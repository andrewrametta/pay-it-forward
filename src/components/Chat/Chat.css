div.chat-box {
  display: flex;
  height: 100%;
}

div.chat-container {
  height: 100%;
  width: 100%;
  display: flex;
  flex-direction: column;
}

div.chat-messages {
  flex: 4;
  overflow: scroll;
}

div.chat-form-box {
  align-self: flex-end;
  display: flex;
  flex-direction: row;
  width: 100%;
}

form.chat-form {
  width: 100%;
  display: flex;
}

input.chat-input {
  margin: 0;
}

button.chat-submit {
  flex: 1;
  margin: 0;
}

div.message-sent {
  background: rgb(48, 0, 96);
  padding: 5px;
  margin-bottom: 5px;
  box-shadow: 2px 2px 6px 0 rgba(41, 2, 80, 0.1);
  border-radius: 10px;
  display: flex;
  flex-direction: row;
}

div.message-recieved {
  background: white;
  padding: 5px;
  margin-bottom: 5px;
  box-shadow: 2px 2px 6px 0 rgba(41, 2, 80, 0.1);
  border-radius: 10px;
  display: flex;
}

p.chat-user-sent {
  color: #aa7dfe;
  margin: 0;
  padding-right: 10px;
  font-weight: bold;
}

p.chat-user-recieved {
  color: rgb(55, 55, 55);
  margin: 0;
  padding-right: 10px;
  font-weight: bold;
}

p.chat-message-sent {
  color: white;
}
p.chat-message-recieved {
  color: #4500c9;
}

p.chat-date-sent {
  color: white;
}

p.chat-date-recieved {
  color: rgb(0, 38, 121);
}
