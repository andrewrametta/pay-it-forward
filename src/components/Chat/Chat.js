import React, { useContext, useState, useEffect, useCallback } from "react";
import AppContext from "../../AppContext";
import AuthAPIService from "../../services/auth-api-service";
import "./Chat.css";

function Chat(props) {
  const conversations_id = props.match.params.conversation_id;
  const [error, setError] = useState("");
  const { messages, setMessages, username } = useContext(AppContext);

  const getDataCallback = useCallback(() => {
    AuthAPIService.getMessage(conversations_id)
      .then((messages) => {
        setMessages(messages);
      })
      .catch((error) => {
        setError(error);
      });
  }, [conversations_id, setMessages]);

  useEffect(() => {
    getDataCallback();
    const timeout = setInterval(getDataCallback, 6000);
    return () => {
      clearInterval(timeout);
    };
  }, [conversations_id, getDataCallback]);

  const handleMessage = (e) => {
    e.preventDefault();
    const { chat } = e.target;
    AuthAPIService.postMessage({
      conversations_id: conversations_id,
      text: chat.value,
      message_status: "not seen",
    })
      .then((message) => {
        setMessages([...messages, message]);
        e.target.reset();
      })
      .catch((res) => {
        console.log(error);
      });
  };

  return (
    <div className="chat-container">
      <div className="chat-messages">
        {messages.map((message, indx) => {
          return message.username === username ? (
            <div className="message-sent" key={indx}>
              <div className="user-info">
                <p className="chat-user-sent">{message.username}</p>
                <p className="chat-date-sent">
                  {new Date(message.timestamp).toLocaleDateString()}
                </p>
              </div>
              <p className="chat-message-sent">{message.text}</p>
            </div>
          ) : (
            <div className="message-recieved" key={indx}>
              <div className="user-info">
                <p className="chat-user-recieved">{message.username} </p>
                <p className="chat-date-recieved">
                  {new Date(message.timestamp).toLocaleDateString()}
                </p>
              </div>
              <p className="chat-message-recieved">{message.text}</p>
            </div>
          );
        })}
      </div>
      {error && <h3>Something went wrong</h3>}
      <div className="chat-form-box">
        <form className="chat-form" onSubmit={handleMessage}>
          <input
            className="chat-input"
            type="text"
            name="chat"
            id="chat"
            placeholder="type a message"
            aria-label="chat-message"
          ></input>
          <button className="chat-submit" type="submit">
            Send
          </button>
        </form>
      </div>
    </div>
  );
}

export default Chat;
