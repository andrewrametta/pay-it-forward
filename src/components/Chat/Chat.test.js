import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import Chat from "./Chat";
// setup file
import { configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import AppContext from "../../AppContext";

configure({ adapter: new Adapter() });

describe(`Chat component`, () => {
  it("renders the complete page", () => {
    const wrapper = shallow(
      <AppContext.Provider>
        <Chat />
      </AppContext.Provider>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
