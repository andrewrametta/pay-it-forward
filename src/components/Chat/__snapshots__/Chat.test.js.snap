// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chat component renders the complete page 1`] = `<Chat />`;
